common/src/main/java/com/wynntils/core/config/Config.java

```diff
@@ -20,4 +20,7 @@
 
     /** Whether this option should be visible to users */
     boolean visible() default true;
+
+    /** Internal configs ignore ConfigHolder's userEdited field and always save */
+    boolean internal() default false;
```

Comment:
maybe instead of requiring this extra flag, `ConfigHolder` should check whether `userEdited` is true _or_ if the field is in a non-default state. previously these two statements were redundant because we only edited fields through the holders, but it would work as a catch-all that doesn't need us to supply this extra info